zingolabs/zingolib
# reorg changes outbound tx height

Closes https://github.com/pacu/zwcd/issues/47 
Closes https://github.com/zingolabs/zingolib/issues/620
### ReOrg Changes Outbound Tx MinedHeight

A Re Org occurs and changes the height of an outbound transaction

Pre-condition: Wallet has funds

Steps:
1. create fake chain
   * 1a. sync to latest height
2. send transaction to recipient address
3. getIncomingTransaction
4. stage transaction at `sentTxHeight`
5. applyHeight(sentTxHeight)
6. sync to latest height
   * 6a. verify that there's a pending transaction with a mined height of sentTxHeight
7. stage 15  blocks from `sentTxHeight`
7. a stage sent tx to `sentTxHeight + 2`
8. `applyHeight(sentTxHeight + 1)` to cause a 1 block reorg
9. sync to latest height
10. verify that there's a pending transaction with -1 mined height
11. `applyHeight(sentTxHeight + 2)`
   * 11a. sync to latest height
12. verify that there's a pending transaction with a mined height of `sentTxHeight + 2`
13. apply height(`sentTxHeight + 15`)
14. sync to latest height
15. verify that there's no pending transaction and that the tx is displayed on the sentTransactions collection
